src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -781,34 +780,29 @@ internal ReadResult GetReadAsyncResult()
                 ThrowHelper.ThrowInvalidOperationException_GetResultNotCompleted();
             }
 
-            var result = new ReadResult();
             lock (_sync)
             {
-                GetReadResult(ref result);
+                GetReadResult(out ReadResult result);
+                return result;
             }
-            return result;
         }
 
-        private void GetReadResult(ref ReadResult result)
+        private void GetReadResult(out ReadResult result)
         {
-            if (_writerCompletion.IsCompletedOrThrow())
-            {
-                result._resultFlags |= ResultFlags.Completed;
-            }
-
-            bool isCanceled = _readerAwaitable.ObserveCancelation();
-            if (isCanceled)
-            {
-                result._resultFlags |= ResultFlags.Canceled;
-            }
-
+            var isCompleted = _writerCompletion.IsCompletedOrThrow();
+            var isCanceled = _readerAwaitable.ObserveCancelation();
+           
             // No need to read end if there is no head
             BufferSegment head = _readHead;
-
             if (head != null)
             {
                 // Reading commit head shared with writer
-                result._resultBuffer = new ReadOnlySequence<byte>(head, _readHeadIndex, _commitHead, _commitHeadIndex - _commitHead.Start);
+                var readOnlySequence = new ReadOnlySequence<byte>(head, _readHeadIndex, _commitHead, _commitHeadIndex - _commitHead.Start);
+                result = new ReadResult(readOnlySequence, isCanceled, isCompleted);
+            }
+            else
+            {
+                result = new ReadResult(new ReadOnlySequence<byte>(), isCanceled, isCompleted);
```

Comment:
default

Reply:
Isn't there any way that `isCanceled` or `isCompleted` is `true`?

Reply:
Think he means `new ReadOnlySequence<byte>()` => `default`

So
```csharp
result = new ReadResult(default, isCanceled, isCompleted);
```

Reply:
Ah, Sorry, that wasn't clear -  that's fine.